docs/DeveloperGuide.md

```diff
@@ -154,90 +155,44 @@ Classes used by multiple components are in the `seedu.address.commons` package.
-than attempting to perform the undo.
-
-</div>
-
-The following sequence diagram shows how the undo operation works:
-
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
-
-</div>
-
-The `redo` command does the opposite — it calls `Model#redoMyInsuRec()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores MyInsuRec to that state.
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `MyInsuRecStateList.size() - 1`, pointing to the latest MyInsuRec state, then there are no undone MyInsuRec states to restore. The `redo` command uses `Model#canRedoMyInsuRec()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
-
-</div>
-
-Step 5. The user then decides to execute the command `listClient`. Commands that do not modify MyInsuRec, such as `listClient`, will usually not call `Model#commitMyInsuRec()`, `Model#undoMyInsuRec()` or `Model#redoMyInsuRec()`. Thus, the `MyInsuRecStateList` remains unchanged.
-
-![UndoRedoState4](images/UndoRedoState4.png)
-
-Step 6. The user executes `clear`, which calls `Model#commitMyInsuRec()`. Since the `currentStatePointer` is not pointing at the end of the `MyInsuRecStateList`, all MyInsuRec states after the `currentStatePointer` will be purged. Reason: It no longer makes sense to redo the `add n/David …​` command. This is the behavior that most modern desktop applications follow.
-
-![UndoRedoState5](images/UndoRedoState5.png)
-
-The following activity diagram summarizes what happens when a user executes a new command:
-
-<img src="images/CommitActivityDiagram.png" width="250" />
-
-#### Design considerations:
-
-**Aspect: How undo & redo executes:**
-
-* **Alternative 1 (current choice):** Saves the entire MyInsuRec.
-  * Pros: Easy to implement.
-  * Cons: May have performance issues in terms of memory usage.
-
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delClient`, just save the client being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
-
-_{more aspects and alternatives to be added}_
-
-### \[Proposed\] Data archiving
-
-_{Explain here how the data archiving feature will be implemented}_
+No need to import handle Pattern, and Matcher classes.
+String matches are easier to understand than regexes.
+- Cons:
 
+Do not offer as much flexibility as regexes.
+List of String returned by getPrefix() cumbersome to use.
 
```

Comment:
I think for the alternative solution, 
you should list down both the current solution and the alt solution.
then bring your rationale about why you chose the current version instead of the alt version down from above.